test/integration-tests/PomaceSettlement.t.sol

```diff
@@ -0,0 +1,210 @@
+
+        amounts[0] = 1 * UNIT;
+        amounts[1] = 1 * UNIT;
+
+        vm.warp(expiry);
+
+        pomace.batchSettleOptions(address(this), ids, amounts);
+
+        assertEq(usdc.balanceOf(address(this)), selfUsdcBefore - 4500 * 1e6);
+        assertEq(usdc.balanceOf(address(engine)), engineUsdcBefore + 4500 * 1e6);
+
+        assertEq(weth.balanceOf(address(this)), 2 * 1e18);
+        assertEq(weth.balanceOf(address(engine)), engineWethBefore - 2 * 1e18);
+    }
+
+    function testSettleDiffCollat() public {
+        weth.approve(address(engine), 1 * 1e18);
+
+        uint256[] memory ids = new uint256[](2);
+        uint256[] memory amounts = new uint256[](2);
+
+        ids[0] = _mintCallOption(1600 * 1e6, wethCollatProductId, 1 * UNIT);
+        ids[1] = _mintPutOption(1900 * 1e6, usdcCollatProductId, 1 * UNIT);
+
+        amounts[0] = 1 * UNIT;
+        amounts[1] = 1 * UNIT;
+
+        vm.warp(expiry);
+
+        pomace.batchSettleOptions(address(this), ids, amounts);
+
+        // receive 1900 usd payout to self (put option) and send 1600 debt (call option) (1900 - 16000 = 300)
+        assertEq(usdc.balanceOf(address(this)), selfUsdcBefore + 300 * 1e6);
+        assertEq(usdc.balanceOf(address(engine)), engineUsdcBefore - 300 * 1e6);
+
+        assertEq(weth.balanceOf(address(this)), 0);
+        assertEq(weth.balanceOf(address(engine)), engineWethBefore);
+    }
+
+    function testSettlePutWithNonUnderlyingNorStrike() public {
+        weth.mint(address(this), 1 * 1e18);
+        weth.approve(address(engine), 1 * 1e18);
+
+        // eth put collateralized in SDYC
+        MockERC20 sdyc = new MockERC20("SDYC", "SDYC", 6);
+
+        vm.label(address(sdyc), "SDYC");
+        sdyc.mint(address(engine), 10_000 * 1e6);
+
+        uint8 sdycId = pomace.registerAsset(address(sdyc));
+
+        pomace.setCollateralizable(address(usdc), address(sdyc), true);
+
+        // create a product with collateral not same as strike
+        uint32 productId = ProductIdUtil.getProductId(engineId, wethId, usdcId, sdycId);
+        uint256 tokenId = _mintPutOption(2000 * 1e6, productId, 1 * UNIT);
+
+        vm.warp(expiry);
+        oracle.setExpiryPrice(address(sdyc), address(usdc), 1.25 * 1e6);
+
+        pomace.settleOption(address(this), tokenId, 1 * UNIT);
+
+        uint256 expectedSdycPayout = 1600 * 1e6;
+        uint256 expectedEngineSdycBalance = 10_000 * 1e6 - expectedSdycPayout;
+
+        assertEq(sdyc.balanceOf(address(this)), expectedSdycPayout);
+        assertEq(sdyc.balanceOf(address(engine)), expectedEngineSdycBalance);
+
+        assertEq(weth.balanceOf(address(this)), 0);
+        assertEq(weth.balanceOf(address(engine)), engineWethBefore + 1 * 1e18);
+
+        assertEq(usdc.balanceOf(address(this)), selfUsdcBefore);
+        assertEq(usdc.balanceOf(address(engine)), engineUsdcBefore);
+    }
+
+    function testSettleCallWithNonUnderlyingNorStrike() public {
```

Comment:
this and `testSettlePutWithNonUnderlyingNorStrike` should be in their own test contract

Reply:
i meant create a new test contract with shared setup, some similarities between these two tests.